client/components/data/query-username-validation/index.jsx

```diff
@@ -0,0 +1,64 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { bindActionCreators } from 'redux';
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { debounce } from 'lodash';
+import debug from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { clearUsernameValidation, validateUsername } from 'state/username/actions';
+
+/**
+ * Module variables
+ */
+const log = debug( 'calypso:query-username-validation' );
+
+class QueryUsernameValidation extends Component {
+	static propTypes = {
+		username: PropTypes.string,
+	};
+
+	componentDidMount() {
+		this.props.clearUsernameValidation();
+		this.debouncedValidateUsername();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.username !== prevProps.username ) {
+			this.debouncedValidateUsername();
+		}
+	}
+
+	componentWillUnmount() {
+		this.props.clearUsernameValidation();
+	}
+
+	debouncedValidateUsername = debounce( this.requestForValidation, 600 );
+
+	requestForValidation() {
+		const { username } = this.props;
+		log( `Validating username ${ username }` );
+		this.props.validateUsername( username );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect( null, dispatch =>
+	bindActionCreators(
```

Comment:
There's a handy shorthand here. You can just pass 
```
{
  clearUsernameValidation,
  validateUsername,
}
```
and it'll auto bind them to `dispatch`